src/code/z_actor.c

```diff
@@ -447,101 +449,128 @@ void Target_Draw(TargetContext* targetCtx, PlayState* play) {
+    // to move the actor off screen, if its even possible.
+    if ((curLockOnActor != NULL) && (targetCtx->reticleSpinCounter == 0)) {
+        Actor_ProjectPos(play, &curLockOnActor->focus.pos, &projectedFocusPos, &cappedInvWDest);
+
         if (((projectedFocusPos.z <= 0.0f) || (1.0f <= fabsf(projectedFocusPos.x * cappedInvWDest))) ||
             (1.0f <= fabsf(projectedFocusPos.y * cappedInvWDest))) {
-            actorArg = NULL;
+            curLockOnActor = NULL;
         }
     }
 
-    if (actorArg != NULL) {
-        if (actorArg != targetCtx->lockOnActor) {
+    if (curLockOnActor != NULL) {
+        if (curLockOnActor != targetCtx->reticleActor) {
             s32 lockOnSfxId;
 
-            Target_InitReticle(targetCtx, actorArg->category, play);
-            targetCtx->lockOnActor = actorArg;
+            // Set up a new lock-on
+            Target_InitReticle(targetCtx, curLockOnActor->category, play);
+            targetCtx->reticleActor = curLockOnActor;
 
-            if (actorArg->id == ACTOR_EN_BOOM) {
+            if (curLockOnActor->id == ACTOR_EN_BOOM) {
+                // Don't draw the reticle when locked onto the boomerang.
+                // Note that it isn't possible to lock onto the boomerang, so this code doesn't do anything.
+                // This implies that the boomerang camera lock may have been implemented with Z-Targeting at one point,
+                // but was eventually implemented as its own camera mode instead.
                 targetCtx->reticleFadeAlphaControl = 0;
             }
 
-            lockOnSfxId = CHECK_FLAG_ALL(actorArg->flags, ACTOR_FLAG_0 | ACTOR_FLAG_2) ? NA_SE_SY_LOCK_ON
-                                                                                       : NA_SE_SY_LOCK_ON_HUMAN;
+            lockOnSfxId = CHECK_FLAG_ALL(curLockOnActor->flags, ACTOR_FLAG_0 | ACTOR_FLAG_2) ? NA_SE_SY_LOCK_ON
+                                                                                             : NA_SE_SY_LOCK_ON_HUMAN;
             Sfx_PlaySfxCentered(lockOnSfxId);
         }
 
-        targetCtx->lockOnPos.x = actorArg->world.pos.x;
-        targetCtx->lockOnPos.y = actorArg->world.pos.y - (actorArg->shape.yOffset * actorArg->scale.y);
-        targetCtx->lockOnPos.z = actorArg->world.pos.z;
+        // Update reticle
+
+        targetCtx->reticlePos.x = curLockOnActor->world.pos.x;
+        targetCtx->reticlePos.y =
+            curLockOnActor->world.pos.y - (curLockOnActor->shape.yOffset * curLockOnActor->scale.y);
+        targetCtx->reticlePos.z = curLockOnActor->world.pos.z;
 
         if (targetCtx->reticleSpinCounter == 0) {
-            f32 temp5 = (500.0f - targetCtx->reticleRadius) * 3.0f;
-            f32 temp6 = (temp5 < 30.0f) ? 30.0f : ((100.0f < temp5) ? 100.0f : temp5);
+            f32 step = (500.0f - targetCtx->reticleRadius) * 3.0f;
+            f32 reticleZoomStep = (step < 30.0f) ? 30.0f : ((100.0f < step) ? 100.0f : step);
```

Comment:
does the `CLAMP` macro match here?

Reply:
I believe it was clamp at some point in the past but it got removed, I suppose to match retail. I can try tho

Reply:
nvm it seems fine